File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int lenT=t.size();
        int lenS=s.size();
        if(lenT<lenS){
            return false;
        }
        int i = 0;
        int j = 0;
        while (i < lenS and j < lenT){
            if (s[i]==t[j]){
                i++;
            }
            j++;
        }
    return (i == lenS);
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = j = 0
        if len(t) < len(s):
            return False
        
        while i < len(t) and j < len(s):
            if t[i] == s[j]:
                j += 1
            i += 1
        

        return j == len(s)
        
        
